modules/rational-defaults.el

```diff
@@ -11,14 +11,44 @@
 
 ;;; Code:
 
+;; Defining variable for XDG directories specifications
+(defvar xdg-data-home-directory
+  (if (getenv "XDG_DATA_HOME")
+    (getenv "XDG_DATA_HOME")
+    (expand-file-name ".local/share/")))
+
+(defvar xdg-cache-home-directory
+  (if (getenv "XDG_DATA_HOME")
+    (getenv "XDG_DATA_HOME")
+    (expand-file-name ".cache/")))
+
+;; Keep emacs folder clean
+
+;; Backup file ( init.el~ )
+(customize-set-variable backup-directory-alist `(("." . ,(expand-file-name "emacs/backups/" xdg-data-home-directory))))
+;; Emacs autosave files ( #init.el# )
+;; The autosave feature doesn't create the directory
+(make-directory (expand-file-name "emacs/auto-save/" xdg-cache-home-directory) t)
+(customize-set-variable auto-save-file-name-transforms
+    `((".*" ,(expand-file-name "emacs/auto-save/" xdg-cache-home-directory) t)))
+
+;; eshell history, lastdir and aliases
+(customize-set-variable eshell-directory-name 
```

Comment:
eshell stuff should probably fit in the `rational-config-var-directory` unless we have a `prefer-xdg-paths` type of setting.